tensorflow/compiler/mlir/hlo/include/mlir-hlo/Dialect/mhlo/IR/hlo_ops.td

```diff
@@ -1651,7 +1660,9 @@ def HLO_ReshapeOp: HLO_Op<"reshape",
   let hasCustomHLOConverter = 1;
 }
 
-def HLO_DynamicReshapeOp: HLO_Op<"dynamic_reshape", [NoSideEffect]> {
+def HLO_DynamicReshapeOp: HLO_Op<"dynamic_reshape", [NoSideEffect,
+    DeclareOpInterfaceMethods<InferShapedTypeOpInterface,
+        ["inferReturnTypeComponents", "reifyReturnTypeShapes"]>]> {
```

Comment:
Can we avoid repeating this boilerplate somehow?

Reply:
One possible solution is that putting these interfaces into the base op and provide a default implementation (e.g. return failure directly). Is there any suggestion about this?